brightlayer-ui/doc-it
# Update dev doc links

Fixes BLUI-3515


#### Changes proposed in this Pull Request:
- Updated all ng and react's storybook links to point to the new dev doc
  - On the component catalog page (talking about http://localhost:3000/development/components), I made everything point to the dev doc's dev branch if the current hostname is not the docit master branch. That is, the docit master points to the dev doc master; other docit branches point to the dev doc dev branch.
  - I thought about doing the same thing on design patterns pages but figured it would cost some effort to implement, so I didn't do it
  - Open to comments on this topic
  - All dev doc links are whitelisted due to SPA's technical limitation. A follow-up story (BLUI-3570) will explore further once we get our own hosting
- Updated all Material UI to MUI
  - Updated MUI links to avoid page redirects in mui.com


#### To Test:
- `yarn start`
- Go to http://localhost:3000/development/components
- Verify that all links are pointing to their appropriate locations on the dev doc
- Go to the design patterns page
- Verify that every component linked on the page bottom is pointing to their appropriate location
